src/docs/asciidoc/index.adoc

```diff
@@ -815,6 +816,257 @@ and it provides tasks to generate a GraalVM native image and also creates the Gr
+
+=== Configuration
+
+The `io.micronaut.aot` plugin is an extension to the `io.micronaut.application` plugin.
+
+[source, groovy, subs="verbatim,attributes", role="multi-language-sample"]
+----
+plugins {
+    ...
+    id "io.micronaut.application" version "{gradle-project-version}"
+    id "io.micronaut.aot" version "{gradle-project-version}"
+    ...
+}
+----
+
+[source, kotlin, subs="verbatim,attributes", role="multi-language-sample"]
+----
+plugins {
+    ...
+    id("io.micronaut.application") version "{gradle-project-version}"
+    id("io.micronaut.aot") version "{gradle-project-version}"
+    ...
+}
+----
+
+This will add an `aot` DSL block to the `micronaut` extension, which can be used to enable optimizations:
+
+[source, groovy, subs="verbatim,attributes", role="multi-language-sample"]
+----
+micronaut {
+    ...
+    aot {
+        // optional, override the Micronaut AOT version
+        version = "1.0.0"
+
+        // optimizations configuration
+        optimizeServiceLoading = true
+        convertYamlToJava = true
+        precomputeOperations = true
+        sealEnvironment = true
+    }
+}
+----
+
+[source, kotlin, subs="verbatim,attributes", role="multi-language-sample"]
+----
+micronaut {
+    ...
+    aot {
+        // optional, override the Micronaut AOT version
+        version.set("1.0.0")
+
+        // optimizations configuration
+        optimizeServiceLoading.set(true)
+        convertYamlToJava.set(true)
+        precomputeOperations.set(true)
+        sealEnvironment.set(true)
+    }
+}
+----
+
+Because Micronaut AOT is an extensible optimization engine, not all optimizations are known beforehand by the plugin, which means that not all of them may be accessible via the DSL.
+For this reason, it is possible to provide a _Micronaut AOT configuration file_ instead:
+
+[source, groovy, subs="verbatim,attributes", role="multi-language-sample"]
+----
+micronaut {
+    ...
+    aot {
+        configFile = file("gradle/micronaut-aot.properties")
+    }
+}
+----
+
+[source, kotlin, subs="verbatim,attributes", role="multi-language-sample"]
+----
+micronaut {
+    ...
+    aot {
+        configFile.set(file("gradle/micronaut-aot.properties"))
+    }
+}
+----
+
+NOTE: You can provide both a configuration file and `aot` DSL optimizations.
+The configuration will be merged, by reading the file first, then using the DSL options.
+
+If you want to know about all possible optimizations, you can run the `createAotSampleConfigurationFiles` which will generate a couple of sample files:
+
+The `build/generated/aot/samples/jit/jit.properties` will contain the optimizations which are relevant to an application running in the regular Java virtual machine, for example:
```

Comment:
It good be nice to show the file created when running the task. Here in the docs it's explained but it is more helpful to have it when running the task. Something lilke:

```
$ gw createAotSampleConfigurationFiles                                  
...

> Task :createAotJitSample
Created build/generated/aot/samples/jit/jit.properties <----- 

> Task :createAotNativeSample
___ And the same here__

> Task :createAotSampleConfigurationFiles

___ And here__
```

Reply:
In general I'm not a big fan of verbose builds. Having something which dumps everything it does. For example `compileJava` doesn't output "wrote class files in ...". Now for this particular case, since it's tasks whose sole goal is to generate a sample, why not. But certainly not duplicated in both the tasks which create the samples and the one which is the lifecycle task :)

Reply:
Yes, I agree that for common tasks it doesn't make sense but for some of them it helps users a lot